File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool check(TreeNode* root,long long minr,long long maxr){
        if(root==nullptr){
            return true;
        }
        if(root->val<=minr || root->val>=maxr){
            return false;
        }
        return 
        (check(root->left,minr,root->val)&&
        check(root->right,root->val,maxr));
    }
    bool isValidBST(TreeNode* root) {
        return check(root,LONG_MIN,LONG_MAX);
        
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def height(self,root):
        if not root:
            return True
        lefth=self.height(root.left)
        righth=self.height(root.right)
        return 1+max(lefth,righth)
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True
        leftht=Solution.height(self,root.left)
        right=Solution.height(self,root.right)
        if abs(leftht-right)>1:
            return False
        return self.isBalanced(root.left)and self.isBalanced(root.right)
        